internal/providers/terraform/azure/log_analytics_workspace.go

```diff
@@ -22,6 +36,14 @@ func newLogAnalyticsWorkspace(d *schema.ResourceData, u *schema.UsageData) *sche
 		sku = d.Get("sku").String()
 	}
 
+	sentinelEnabled := logAnalyticsWorkspaceDetectSentinel(d)
+	if !sentinelEnabled {
+		logSolutionRefs := d.References("azurerm_log_analytics_solution.workspace_resource_id")
+		if len(logSolutionRefs) > 0 {
+			sentinelEnabled = logAnalyticsWorkspaceDetectSentinel(logSolutionRefs[0])
+		}
+	}
```

Comment:
this seems a bit weird to me - can we just move this inside the `logAnalyticsWorkspaceDetectSentinel`

Reply:
There are two ways to search for the data connector resources - directly on workspace or through analytics solution resource. The idea was to search on `BuildResource` level. If we put this block in the function as well, we'll have to either duplicate the `for` loop over `sentinelDataConnectorRefs` or call itself recursively (which is okay as it will be called just once), but that part might be confusing as well:

```go
sentinelEnabled := logAnalyticsWorkspaceDetectSentinel(d)

// ...

func logAnalyticsWorkspaceDetectSentinel(d *schema.ResourceData) bool {
	for _, ref := range sentinelDataConnectorRefs {
		if len(d.References(ref)) > 0 {
			return true
		}
	}

	logSolutionRefs := d.References("azurerm_log_analytics_solution.workspace_resource_id")
	if len(logSolutionRefs) > 0 {
		return logAnalyticsWorkspaceDetectSentinel(logSolutionRefs[0])
		// for _, ref := range sentinelDataConnectorRefs {
		// 	if len(logSolutionRefs[0].References(ref)) > 0 {
		// 		return true
		// 	}
		// }
	}


	return false
}
```

I'll use recursion option 👍 .